virtcontainers/kata_agent.go

```diff
@@ -132,6 +129,40 @@ const (
 	grpcStopTracingRequest       = "grpc.StopTracingRequest"
 )
 
+// The function is declared this way for mocking in unit tests
+var kataHostSharedDir = func() string {
+	path := "/run/kata-containers/shared/sandboxes/"
```

Comment:
I think we need to find another solution here as we shouldn't be hard-coding paths like this in non-test code. Maybe add some sort of test setup function? Same comment for functions below.

Reply:
It seems that today its hardcoded in this same file already, right? See https://github.com/kata-containers/runtime/blob/master/virtcontainers/kata_agent.go#L59-L62

WDYT @jodh-intel ?


How are these handled/specified today?  Seems like the initial path assignment is problematic, and we should be using

Reply:
Yes, we already have `kataHostSharedDir` and `kataGuestSharedDir` globals for these paths, so why can't the mocking code either use or change those variables for testing purposes? I don't understand the need for the duplication - added to which it looks very odd to me.

Any thoughts @marcov?

Reply:
The comment on the variables is wrong. The reason to have functions in place of const is to have those directories placed in a path accessible by regular users when running rootless.

As alternative, we can leave the const declaration as before, and wherever they are used do something like:
```golang
rootless.ResolvePath(kataHostSharedDir)
```
But I like more how it is done now.